syntax = "proto3";

package poker;

import "cards.proto";

enum HandType {
  HANDTYPE_UNSPECIFIED = 0;
  HIGH_CARD = 1;
  ONE_PAIR = 2;
  TWO_PAIR = 3;
  THREE_OF_A_KIND = 4;
  STRAIGHT = 5;
  FLUSH = 6;
  FULL_HOUSE = 7;
  FOUR_OF_A_KIND = 8;
  STRAIGHT_FLUSH = 9;
}

enum HandAttribute {
  HAND_ATTR_UNSPECIFIED = 0;
  HAND_ATTR_ONE_GAP_CONNECTOR = 0x01;
  HAND_ATTR_TWO_GAP_CONNECTOR = 0x02;
  HAND_ATTR_THREE_GAP_CONNECTOR = 0x04;
  HAND_ATTR_FLUSH_DRAW = 0x08;
  HAND_ATTR_BACKDOOR_FLUSH_DRAW = 0x10;
  HAND_ATTR_OUTSIDE_STRAIGHT_DRAW = 0x20;
  HAND_ATTR_INSIDE_STRAIGHT_DRAW = 0x40;
  HAND_ATTR_DOUBLE_INSIDE_STRAIGHT_DRAW = 0x80;
}

message Hand {
  HandType type = 1;
  repeated Rank rank = 2;
  int32 sort_code = 3;
}
